pandas/core/indexes/base.py

```diff
@@ -4930,6 +4936,18 @@ def _get_leaf_sorter(labels: list[np.ndarray]) -> npt.NDArray[np.intp]:
         )
         return join_index, left_indexer, right_indexer
 
+    def _can_join_monotonic(self, other):
+        if (
+            # not isinstance(self.dtype, CategoricalDtype)
```

Comment:
Nit: Is this still relevant?

Reply:
nope, forgot to remove that. Its now removed. thanks